pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -62,7 +62,18 @@ def resume_start(self) -> None:
             return
 
         rank_zero_info(f"Restoring states from the checkpoint path at {checkpoint_path}")
-        self._loaded_checkpoint = self.trainer.training_type_plugin.load_checkpoint(checkpoint_path)
+        self._loaded_checkpoint = self._load_and_validate_checkpoint(checkpoint_path)
+
+    def _load_and_validate_checkpoint(self, checkpoint_path: _PATH) -> Optional[Dict[str, Any]]:
```

Comment:
let's use this here too: https://github.com/PyTorchLightning/pytorch-lightning/blob/ec828b826717cd3b5beabcb6d0cacf41b2320a98/pytorch_lightning/trainer/connectors/checkpoint_connector.py#L147-L149